packages/docusaurus-theme-common/src/utils/useChangeRoute.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {useRef, useEffect} from 'react';
+import {useLocation} from '@docusaurus/router';
+
+export function useChangeRoute(onRouteChange: () => void): void {
+  const {pathname} = useLocation();
+  const initialPathname = useRef(pathname);
+
+  useEffect(() => {
+    if (pathname !== initialPathname.current) {
```

Comment:
That seems to work on most cases but what if user is on `/a`, navigates to `/b`, and then navigates back to `/a`? 

I think you should rather use a hook like [usePrevious](https://usehooks.com/usePrevious/) to make sure that the callback fires even if the user goes back to the page that was initially loaded

Reply:
Otherwise that looks like to fix the problem! thanks

Reply:
Could you reproduce this issue on the preview site? Currently, when navigating back to initially loaded page, `useEffect` fires as expected.

Reply:
Oh I'm sorry, didn't notice that you reassigned `initialPathname` to latest route.

So somehow you already handle that case, it's just that I'm not fan of the naming "initialPathname" considering its value is changed over time. Will change change the name and merge